x/programs/rust/wasmlanche_sdk/src/state.rs

```diff
@@ -77,3 +78,40 @@ impl State {
         from_slice(&val).map_err(|_| StateError::InvalidBytes)
     }
 }
+
+/// Converts a raw pointer to a deserialized value.
+/// Expects the first 4 bytes of the pointer to represent the [length] of the serialized value,
+/// with the subsequent [length] bytes comprising the serialized data.
+// pub fn from_raw_ptr<V>(ptr: i64) -> V
+// where
+//     V: serde::de::DeserializeOwned,
+// {
+//     let (bytes, _) = bytes_and_length(ptr);
+//     from_slice(&bytes).expect("failed to deserialize")
+// }
```

Comment:
might warrant a discussion about using a serialization package. Would make the code cleaner and have more out of the box functionality

Reply:
I think we agreed on Borsh for now. @hexfusion?

Reply:
Yes 👍

Reply:
using borsh now 🙌